dcc-portal-ui/app/scripts/app/js/app.js

```diff
@@ -469,24 +470,40 @@
       }]);
 
     })
-    .run(function($state, $location, $window, $timeout, $rootScope, cfpLoadingBar, HistoryManager, gettextCatalog, Settings) {
+    .run(function($state, $location, $window, $timeout, $rootScope, cfpLoadingBar, HistoryManager, gettextCatalog, Settings, REDIRECTS) {
       
       // Setting the initial language to English CA.
       gettextCatalog.setCurrentLanguage('en_CA');
 
       HistoryManager.addToIgnoreScrollResetWhiteList(['analysis','advanced', 'compound', 'dataRepositories', 'donor', 'beacon', 'project', 'gene']);
       
       $rootScope.$on('$stateChangeError', function(event, toState, toParams, fromState, fromParams, error) {
-        if(error.status === 404){
-          $state.go('404', {page: toState.name, id: toParams.id, url: toState.url}, {location: false});
+        if(error.status === 404) {
+          const redirect = _.find(REDIRECTS, (redirect) => redirect.from === $location.url());
+          let state = '404', page = toState.name;
+
+          if(redirect){
+            state = '301';
+            page = redirect.to;
+          }
+      
+          $state.go(state, {page, id: toParams.id, url: toState.url}, {location: false});
         } else {
           console.error(error.message);
           console.error(error.stack);
         }
       });
 
       $rootScope.$on('$stateNotFound', function() {
-        $state.go('404', {}, {location: false});
+        const redirect = _.find(REDIRECTS, (redirect) => redirect.from === $location.url());
```

Comment:
Can you give me an example? I need something that I can inject inside `app.config`